src/engine/Engine.cpp

```diff
@@ -1192,6 +1200,44 @@ void Engine::performMILPSolverBoundedTightening()
     }
 }
 
+void Engine::performMILPSolverBoundedTighteningForSingleLayer( unsigned targetIndex )
+{
+    if ( _networkLevelReasoner && Options::get()->gurobiEnabled() && !Options::get()->getBool( Options::SKIP_LP_TIGHTENING_AFTER_SPLIT ) )
```

Comment:
Add another check condition
Options::get()->getMILPSolverBoundTighteningType() != MILPSolverBoundTighteningType::NONE

Reply:
Since the Options object is not thread safe and there is a chance that multiple Engine object be accessing the Options object.  Maybe store  `Options::get()->gurobiEnabled(),  Options::get()->getBool( Options::SKIP_LP_TIGHTENING_AFTER_SPLIT ) and Options::get()->getMILPSolverBoundTighteningType()` as local variables in the constructor of Engine()

Reply:
Done.